omc3/optics_measurements/rdt.py

```diff
@@ -267,10 +268,14 @@ def fitting(x, f):
         return f * x
 
     for i, bpm_rdt_data in enumerate(line_amp):
-        popt, pcov = curve_fit(fitting, kick_data, bpm_rdt_data, p0=guess[i])
-        amps[i] = popt[0]
-        sqrt_pcov = np.sqrt(pcov).flat[0]
-        err_amps[i] = sqrt_pcov if np.isfinite(sqrt_pcov) else 0. # if single file is used, the error is reported as Inf, which is then overwritten with 0
+        # Catch the potential 'OptimizeWarning: Covariance of the parameters could not be estimated'
+        # that can happen during the curve_fit
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", OptimizeWarning)
+            popt, pcov = curve_fit(fitting, kick_data, bpm_rdt_data, p0=guess[i])
+            amps[i] = popt[0]
+            sqrt_pcov = np.sqrt(pcov).flat[0]
+            err_amps[i] = sqrt_pcov if np.isfinite(sqrt_pcov) else 0. # if single file is used, the error is reported as Inf, which is then overwritten with 0
```

Comment:
**Note**: Same here, we might want to keep.

Reply:
same comment